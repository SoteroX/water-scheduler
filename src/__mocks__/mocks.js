export const mockEvents = [
  {
    id: 0,
    title: "All Day Event very long title",
    start: new Date(2015, 3, 0),
    end: new Date(2015, 3, 1)
  },
  {
    id: 1,
    title: "Long Event",
    start: new Date(2015, 3, 7),
    end: new Date(2015, 3, 10)
  }
];

export const mockEmptyEvents = [];

export const mockPlants = [
  { name: "Fiddle Leaf Fig", water_after: "7 days" },
  { name: "Croton", water_after: "7 days" },
  { name: "Spider Plant", water_after: "7 days" },
  { name: "Snake Plant", water_after: "14 days" },
  { name: "Money Tree", water_after: "14 days" },
  { name: "Dracaena", water_after: "14 days" },
  { name: "Jade", water_after: "14 days" },
  { name: "Bird's Nest Fern", water_after: "3 days" },
  { name: "Bell Pepper Plant", water_after: "3 days" },
  { name: "Strawberry Plant", water_after: "3 days" },
  { name: "Wavy Fern", water_after: "2 days" }
];

export const mockSortedPlants = {
  "7 days": [
    { name: "Fiddle Leaf Fig", water_after: "7 days" },
    { name: "Croton", water_after: "7 days" },
    { name: "Spider Plant", water_after: "7 days" }
  ],
  "14 days": [
    { name: "Snake Plant", water_after: "14 days" },
    { name: "Money Tree", water_after: "14 days" },
    { name: "Dracaena", water_after: "14 days" },
    { name: "Jade", water_after: "14 days" }
  ],
  "3 days": [
    { name: "Bird's Nest Fern", water_after: "3 days" },
    { name: "Bell Pepper Plant", water_after: "3 days" },
    { name: "Strawberry Plant", water_after: "3 days" }
  ],
  "2 days": [{ name: "Wavy Fern", water_after: "2 days" }]
};

export const mockEmptyPlants = {};

export const mockLists = [
  {
    name: "Fiddle Leaf Fig",
    water_after: "7 days",
    dates: [
      { month: 11, day: 16, year: 2019 },
      { month: 11, day: 23, year: 2019 },
      { month: 11, day: 30, year: 2019 },
      { month: 0, day: 6, year: 2020 }
    ]
  },
  {
    name: "test plant",
    water_after: "14 days",
    dates: [
      { month: 11, day: 16, year: 2019 },
      { month: 11, day: 23, year: 2019 },
      { month: 11, day: 30, year: 2019 },
      { month: 0, day: 6, year: 2020 }
    ]
  }
];
