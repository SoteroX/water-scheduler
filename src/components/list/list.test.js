import React from "react";
import { shallow } from "enzyme";
// Import mocks
import { mockLists } from "../../__mocks__/mocks";
// Import components
import ListContainer from "./list.container";
import List from "./list.component";

describe("List container", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ListContainer />);
  });
  // Makes sure ListContainer component is rendered
  it("renders ListContainer without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("List component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<List list={mockLists} />);
  });
  // Makes sure List component is rendered
  it("renders List component without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
