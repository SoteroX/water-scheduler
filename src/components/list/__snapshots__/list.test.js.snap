// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List component without crashing 1`] = `
<div
  className="list-section"
>
  <h1
    className="list-component-title"
  >
    List Component
  </h1>
  <div
    className="list-ui-container"
  >
    <div
      key="Fiddle Leaf Fig"
    >
      <h3
        className="list-plant-title"
      >
        Fiddle Leaf Fig
      </h3>
      <h5
        className="list-plant-dates"
      >
        Every 
        7 days
      </h5>
      <ul>
        <li
          key="a932b63b-bfdd-4aa2-9af9-9468c079f3d7"
        >
          Mon Dec 16 2019
        </li>
        <li
          key="07108a9e-2cb6-40bb-b9a8-358c2d104bff"
        >
          Mon Dec 23 2019
        </li>
        <li
          key="fdc7382c-0718-42b5-b102-2676c5e19cd9"
        >
          Mon Dec 30 2019
        </li>
        <li
          key="64f8c166-186f-4830-842c-c42e7e91b89f"
        >
          Mon Jan 06 2020
        </li>
      </ul>
    </div>
    <div
      key="test plant"
    >
      <h3
        className="list-plant-title"
      >
        test plant
      </h3>
      <h5
        className="list-plant-dates"
      >
        Every 
        14 days
      </h5>
      <ul>
        <li
          key="1487e092-8735-45aa-81de-2fdafe32c262"
        >
          Mon Dec 16 2019
        </li>
        <li
          key="e1bb9b0a-ead6-4c53-970a-80ff742f7a7e"
        >
          Mon Dec 23 2019
        </li>
        <li
          key="b2eb13b7-02dd-41ac-bd3f-c1f7f45436ef"
        >
          Mon Dec 30 2019
        </li>
        <li
          key="b013acaa-1416-490b-ab96-19db58b013f0"
        >
          Mon Jan 06 2020
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`List container renders ListContainer without crashing 1`] = `
<div>
  <div>
    List couldn't be displayed, please check for errors
  </div>
</div>
`;
